Automattic/notifications-panel
# Upgrade notifications-panel to React 16

While trying to upgrade Calypso to React 16, the first thing that blew up at runtime was `notifications-panel`. That was because of `React.PropTypes`.  I figured while here I might as well upgrade `notifications-panel` to 16 (will also ensure I don't miss anything).


Changelist:
----

1. changes all `React.PropTypes` --> the one found in the separate `prop-types` package using the [official prop-types codemod](https://github.com/reactjs/react-codemod#react-proptypes-to-prop-types).
2. changes all `React.createClass` calls to either `createReactClass` or es6 classes using the [official class codemod](https://github.com/reactjs/react-codemod#explanation-of-the-new-es2015-class-transform-with-property-initializers). 
3. update react-redux, react, and react-dom in `package.json`.
4. profit

To Test
------
1. `rm -rf node_modules`
2. `npm install`
3. `npm start`
4. ensure notifications-panel boots as expected and do a short smoke test.
5. ~~also ensure the new bundle sizes are much smaller than the old ones~~
